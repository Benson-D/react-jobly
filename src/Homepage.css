.Homepage {
  height: 4rem;
  background-color: snow;
}

h1 {
  padding-top: 10px;
}
