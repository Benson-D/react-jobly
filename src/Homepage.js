import React from "react";

/** Renders Homepage
 *
 * Props: none,
 * State: none
 *
 * Routes -> Homepage
 */

function Homepage() {
  return <h1>Welcome To Jobly!</h1>;
}

export default Homepage;
