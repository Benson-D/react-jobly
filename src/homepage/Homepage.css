.Homepage {
  height: calc(40vh - 110px);
  background-color: snow;
  border-bottom: 2px solid hsla(0, 0%, 3%, 0.125);
  text-shadow: 4px 0px 6px rgba(132, 132, 132, 0.74);
}

h1 {
  padding-top: 10px;
  font-size: 38px;
}
