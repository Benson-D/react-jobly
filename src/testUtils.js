import React from "react";
import UserContext from "./auth/UserContext";

const testUser = {
    username: "testuser",
    first_name: "testFirst",
    last_name: "testLast",
    email: "test@gmail.com",
}

const UserProvider =
    ({ children, currUser = testUser, handleAppliedtoJob = () => false }) => (
    <UserContext.Provider value={{ currUser, handleAppliedtoJob }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
