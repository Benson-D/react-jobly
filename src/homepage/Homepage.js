import { useContext } from "react";
import "./Homepage.css";
import UserContext from "../auth/UserContext";

/** Renders Homepage
 *
 * Props: none,
 * State: none
 *
 * Routes -> Homepage
 */

function Homepage() {
  const { currUser } = useContext(UserContext);

  return (
    <main className="Homepage">
      <div className="container text-center">
        <h1 className="mb-4 fw-bold">Jobly</h1>
        <p>A convenient but helpful way to jump start your career</p>
        {currUser && (
          <p className="Homepage-welcome lead">Welcome Back, {currUser.firstName}!</p>
        )}
      </div>
    </main>
  );
}

export default Homepage;
