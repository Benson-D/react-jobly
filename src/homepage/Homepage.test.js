import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Homepage from "./Homepage";
import { UserProvider } from "../testUtils";


it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <Homepage />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders the homepage", function () {
    render(
        <MemoryRouter>
            <UserProvider>
                <Homepage/>
            </UserProvider>
        </MemoryRouter>
    )

    expect(screen.getByText(/Welcome Back/i)).toBeInTheDocument()
})

it("renders the homepage when logged out", function () {
    render(
        <MemoryRouter>
            <UserProvider currUser={null}>
                <Homepage/>
            </UserProvider>
        </MemoryRouter>
    )

    expect(screen.queryByText("Welcome Back")).not.toBeInTheDocument()
})