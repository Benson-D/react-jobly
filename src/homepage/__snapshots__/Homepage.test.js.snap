// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="Homepage"
  >
    <div
      class="container text-center"
    >
      <h1
        class="mb-4 fw-bold"
      >
        Jobly
      </h1>
      <p>
        A convenient but helpful way to jump start your career
      </p>
      <p
        class="Homepage-welcome lead"
      >
        Welcome Back, !
      </p>
    </div>
  </main>
</DocumentFragment>
`;
